diff -crB alpha_dsk_c/src/dsklib_c/mkprodct.csh alpha_dsk_c/src_new/dsklib_c/mkprodct.csh
*** alpha_dsk_c/src/dsklib_c/mkprodct.csh	2014-07-18 18:47:52.000000000 -0700
--- alpha_dsk_c/src_new/dsklib_c/mkprodct.csh	2015-03-26 14:57:41.738345707 -0700
***************
*** 225,232 ****
     echo "      Inserting objects in the library $LIBRARY ..."
     ar  crv $LIBRARY.a *.o
     ranlib  $LIBRARY.a
-    \rm                *.o    
     echo " "
  
  endif
  
--- 225,235 ----
     echo "      Inserting objects in the library $LIBRARY ..."
     ar  crv $LIBRARY.a *.o
     ranlib  $LIBRARY.a
     echo " "
+    echo " Creating dynamic library "
+    set SOFLAGS = `echo $TKCOMPILEOPTIONS | sed 's/-c / /g'`
+    $TKCOMPILER $LDFLAGS $SOFLAGS -shared -Wl,-soname,libcdsk.so -o ../../lib/libcdsk.so *.o
+    \rm                *.o    
  
  endif
  
