From a092f1f86aada11a5b27b15b1481e37e11ff0fb7 Mon Sep 17 00:00:00 2001
From: Mike Lundy <mike@fluffypenguin.org>
Date: Thu, 9 Dec 2010 17:46:34 -0800
Subject: [PATCH] fix jpeg detection

---
 configure.in |   16 +++-------------
 1 files changed, 3 insertions(+), 13 deletions(-)

diff --git a/configure.in b/configure.in
index 4b69da6..7597709 100644
--- a/configure.in
+++ b/configure.in
@@ -897,19 +897,9 @@ elif test "$with_jpeg" = "yes" -o "$with_jpeg" = "" ; then
   fi
 
   if test "$JPEG_SETTING" = "external" -a "$TIFF_SETTING" = "internal" ; then
-    AC_MSG_CHECKING([for width_in_blocks in jpeglib.h])
-
-    rm -f conftest.c
-    echo '#include <stdio.h>' >> conftest.c
-    echo '#include "jpeglib.h"' >> conftest.c
-    echo 'int main() { jpeg_component_info *comptr=0; int i; i = comptr->width_in_blocks; }' >> conftest.c
-    if test -z "`${CC} -o conftest conftest.c 2>&1`" ; then
-      AC_MSG_RESULT([yes])
-    else
-      AC_MSG_RESULT([no])
-      JPEG_SETTING=internal
-    fi
-    rm -f conftest*
+    AC_CHECK_MEMBER([jpeg_component_info.width_in_blocks], [], [JPEG_SETTING=internal],[[
+        #include <stdio.h>
+        #include <jpeglib.h>]])
   fi
 
   if test "$JPEG_SETTING" = "external" ; then   
-- 
1.7.3.2

