diff -crB ./configure ../qt-everywhere-opensource-src-4.8.2_mod/configure
*** ./configure	2012-04-26 12:46:24.000000000 -0700
--- ../qt-everywhere-opensource-src-4.8.2_mod/configure	2012-06-14 09:19:15.000000000 -0700
***************
*** 3437,3447 ****
  
  # auto-detect support for -Xarch on the mac
  if [ "$PLATFORM_MAC" = "yes" ] && [ "$CFG_MAC_XARCH" = "auto" ]; then
!     if "$mactests/xarch.test" "$TEST_COMPILER" "$OPT_VERBOSE" "$mactests" ; then
!         CFG_MAC_XARCH=no
!     else
!         CFG_MAC_XARCH=yes
!     fi
  fi
  
  # don't autodetect support for separate debug info on objcopy when
--- 3437,3443 ----
  
  # auto-detect support for -Xarch on the mac
  if [ "$PLATFORM_MAC" = "yes" ] && [ "$CFG_MAC_XARCH" = "auto" ]; then
!     CFG_MAC_XARCH=no
  fi
  
  # don't autodetect support for separate debug info on objcopy when
***************
*** 4944,4970 ****
              EXTRA_CXXFLAGS="$EXTRA_CXXFLAGS \$(CARBON_CFLAGS)"
              EXTRA_OBJS="qsettings_mac.o qcore_mac.o"
              EXTRA_SRCS="\"$relpath/src/corelib/io/qsettings_mac.cpp\" \"$relpath/src/corelib/kernel/qcore_mac.cpp\""
- 	    if echo "$CFG_MAC_ARCHS" | grep x86 > /dev/null 2>&1; then # matches both x86 and x86_64
- 		X86_CFLAGS="-arch i386"
- 		X86_LFLAGS="-arch i386"
- 		EXTRA_CFLAGS="$X86_CFLAGS $EXTRA_CFLAGS"
- 		EXTRA_CXXFLAGS="$X86_CFLAGS $EXTRA_CXXFLAGS"
-                 EXTRA_LFLAGS="$EXTRA_LFLAGS $X86_LFLAGS"
-             fi
- 	    if echo "$CFG_MAC_ARCHS" | grep ppc > /dev/null 2>&1; then # matches both ppc and ppc64
- 		PPC_CFLAGS="-arch ppc"
- 		PPC_LFLAGS="-arch ppc"
- 		EXTRA_CFLAGS="$PPC_CFLAGS $EXTRA_CFLAGS"
- 		EXTRA_CXXFLAGS="$PPC_CFLAGS $EXTRA_CXXFLAGS"
-                 EXTRA_LFLAGS="$EXTRA_LFLAGS $PPC_LFLAGS"
-             fi
- 	    if [ '!' -z "$CFG_SDK" ]; then
- 		echo "SDK_LFLAGS =-Wl,-syslibroot,$CFG_SDK" >>"$mkfile"
- 		echo "SDK_CFLAGS =-isysroot $CFG_SDK" >>"$mkfile"
- 		EXTRA_CFLAGS="$EXTRA_CFLAGS \$(SDK_CFLAGS)"
- 		EXTRA_CXXFLAGS="$EXTRA_CXXFLAGS \$(SDK_CFLAGS)"
- 		EXTRA_LFLAGS="$EXTRA_LFLAGS \$(SDK_LFLAGS)"
-             fi
          fi
          [ "$CFG_EMBEDDED" != "no" ] && EXTRA_CFLAGS="$EXTRA_CFLAGS -DQWS"
          if [ '!' -z "$D_FLAGS" ]; then
--- 4940,4945 ----
***************
*** 7945,7988 ****
      QMAKE_CONFIG="$QMAKE_CONFIG exceptions_off"
  fi

- # On Mac, set the minimum deployment target for the different architechtures 
- # using the Xarch compiler option when supported (10.5 and up).  On 10.4 the
- # deployment version is set to 10.4 globally using the QMAKE_MACOSX_DEPLOYMENT_TARGET
- # env. variable.
- if [ "$PLATFORM_MAC" = "yes" ] && [ "$CFG_MAC_XARCH" != "no" ]; then
-     case "$PLATFORM" in
-     *macx-clang-libc++)
-         # Avoid overriding the default settings when building clang/libc++.
-         ;;
-     *)
-         if echo "$CFG_MAC_ARCHS" | grep '\<x86\>' > /dev/null 2>&1; then
-             QMakeVar add QMAKE_CFLAGS "-Xarch_i386 -mmacosx-version-min=10.4"
-             QMakeVar add QMAKE_CXXFLAGS "-Xarch_i386 -mmacosx-version-min=10.4"
-             QMakeVar add QMAKE_LFLAGS "-Xarch_i386 -mmacosx-version-min=10.4"
-             QMakeVar add QMAKE_OBJECTIVE_CFLAGS_X86 "-arch i386 -Xarch_i386 -mmacosx-version-min=10.4"
-         fi
-         if echo "$CFG_MAC_ARCHS" | grep '\<ppc\>' > /dev/null 2>&1; then
-             QMakeVar add QMAKE_CFLAGS "-Xarch_ppc -mmacosx-version-min=10.4"
-             QMakeVar add QMAKE_CXXFLAGS "-Xarch_ppc -mmacosx-version-min=10.4"
-             QMakeVar add QMAKE_LFLAGS "-Xarch_ppc -mmacosx-version-min=10.4"
-             QMakeVar add QMAKE_OBJECTIVE_CFLAGS_PPC "-arch ppc -Xarch_ppc -mmacosx-version-min=10.4"
-         fi
-         if echo "$CFG_MAC_ARCHS" | grep '\<x86_64\>' > /dev/null 2>&1; then
-             QMakeVar add QMAKE_CFLAGS "-Xarch_x86_64 -mmacosx-version-min=10.5"
-             QMakeVar add QMAKE_CXXFLAGS "-Xarch_x86_64 -mmacosx-version-min=10.5"
-             QMakeVar add QMAKE_LFLAGS "-Xarch_x86_64 -mmacosx-version-min=10.5"
-             QMakeVar add QMAKE_OBJECTIVE_CFLAGS_X86_64 "-arch x86_64 -Xarch_x86_64 -mmacosx-version-min=10.5"
-         fi
-         if echo "$CFG_MAC_ARCHS" | grep '\<ppc64\>' > /dev/null 2>&1; then
-             QMakeVar add QMAKE_CFLAGS "-Xarch_ppc64 -mmacosx-version-min=10.5"
-             QMakeVar add QMAKE_CXXFLAGS "-Xarch_ppc64 -mmacosx-version-min=10.5"
-             QMakeVar add QMAKE_LFLAGS "-Xarch_ppc64 -mmacosx-version-min=10.5"
-             QMakeVar add QMAKE_OBJECTIVE_CFLAGS_PPC_64 "-arch ppc64 -Xarch_ppc64 -mmacosx-version-min=10.5"
-         fi
-         ;;
-     esac
- fi
- 
  #-------------------------------------------------------------------------------
  # generate QT_BUILD_KEY
  #-------------------------------------------------------------------------------
--- 7920,7925 ----
***************
*** 9258,9260 ****
--- 9202,9210 ----
  echo
  echo To reconfigure, run \'$MAKE confclean\' and \'configure\'.
  echo
+ 
+ echo CFG_ARCH is $CFG_ARCH
+ echo CFG_HOST_ARCH is $CFG_HOST_ARCH
+ echo CFG_MAC_ARCHS is $CFG_MAC_ARCHS
+ echo CFG_MAC_XARCH is $CFG_MAC_XARCH
+ echo QMAKE_FLAGS is $QMAKE_LFLAGS
\ No newline at end of file
Only in ../qt-everywhere-opensource-src-4.8.2_mod/: configure.orig
Only in ../qt-everywhere-opensource-src-4.8.2_mod/: configure~
