--- qtbase/configure	2017-01-27 21:12:18.358644496 -0800
+++ qtbase_new/configure	2017-01-27 21:10:28.760058291 -0800
@@ -3585,19 +3585,7 @@
    CFG_STACK_PROTECTOR_STRONG=no
 fi
 
-# detect the availability of the -Bsymbolic-functions linker optimization
-if [ "$CFG_REDUCE_RELOCATIONS" != "no" ]; then
-    if "$unixtests/bsymbolic_functions.test" "$TEST_COMPILER" "$OPT_VERBOSE"; then
-       if [ "$CFG_REDUCE_RELOCATIONS" = "yes" ]; then
-           echo "-reduce-relocations was requested but this compiler does not support it"
-           echo "Re-run configure with -v for more information"
-           exit 1
-       fi
-        CFG_REDUCE_RELOCATIONS=no
-    else
-        CFG_REDUCE_RELOCATIONS=yes
-    fi
-fi
+CFG_REDUCE_RELOCATIONS=no
 
 # auto-detect GNU make support
 if [ "$CFG_USE_GNUMAKE" = "auto" ] && "$MAKE" -v | grep "GNU Make" >/dev/null 2>&1; then
