diff -ruN OpenSceneGraph-3.0.1/CMakeLists.txt OpenSceneGraph-3.0.1_prior/CMakeLists.txt
--- OpenSceneGraph-3.0.1/CMakeLists.txt	2012-11-25 07:37:25.000000000 -0800
+++ OpenSceneGraph-3.0.1_prior/CMakeLists.txt	2012-11-25 07:29:04.000000000 -0800
@@ -505,12 +505,12 @@
 IF   (BUILD_OSG_EXAMPLES AND NOT ANDROID)
 
 
-    FIND_PACKAGE(FLTK)
-    FIND_PACKAGE(GLUT)
-    FIND_PACKAGE(FOX)
+    MACRO_OPTIONAL_FIND_PACKAGE(FLTK)
+    MACRO_OPTIONAL_FIND_PACKAGE(GLUT)
+    MACRO_OPTIONAL_FIND_PACKAGE(FOX)
 
     SET(wxWidgets_USE_LIBS base core gl net)
-    FIND_PACKAGE(wxWidgets)
+    MACRO_OPTIONAL_FIND_PACKAGE(wxWidgets)
 
 ENDIF(BUILD_OSG_EXAMPLES AND NOT ANDROID)
 
@@ -628,23 +628,23 @@
 # can use Quicktime.
 IF(NOT ANDROID)
     IF(NOT APPLE)
-        FIND_PACKAGE(GIFLIB)
-        FIND_PACKAGE(JPEG)
-        FIND_PACKAGE(PNG)
-        FIND_PACKAGE(TIFF)
+        MACRO_OPTIONAL_FIND_PACKAGE(GIFLIB)
+        MACRO_OPTIONAL_FIND_PACKAGE(JPEG)
+        MACRO_OPTIONAL_FIND_PACKAGE(PNG)
+        MACRO_OPTIONAL_FIND_PACKAGE(TIFF)
         # QuickTime is required for OS X, but optional for Windows.
         IF(WIN32)
-            FIND_PACKAGE(QuickTime)
+            MACRO_OPTIONAL_FIND_PACKAGE(QuickTime)
         ENDIF()
 
     ELSE()
-        FIND_PACKAGE(TIFF)
-        FIND_PACKAGE(QuickTime)
-        FIND_PACKAGE(QTKit)
-        FIND_PACKAGE(CoreVideo)
-        FIND_PACKAGE(CoreMedia)
-        FIND_PACKAGE(QuartzCore)
-        FIND_PACKAGE(AVFoundation)
+        MACRO_OPTIONAL_FIND_PACKAGE(TIFF)
+        MACRO_OPTIONAL_FIND_PACKAGE(QuickTime)
+        MACRO_OPTIONAL_FIND_PACKAGE(QTKit)
+        MACRO_OPTIONAL_FIND_PACKAGE(CoreVideo)
+        MACRO_OPTIONAL_FIND_PACKAGE(CoreMedia)
+        MACRO_OPTIONAL_FIND_PACKAGE(QuartzCore)
+        MACRO_OPTIONAL_FIND_PACKAGE(AVFoundation)
     ENDIF()
 ENDIF()
 
--- OpenSceneGraph-3.2.0/src/osgPlugins/CMakeLists.txt  2016-04-30 14:44:37.972022912 -0700
+++ OpenSceneGraph-3.2.0_prior/src/osgPlugins/CMakeLists.txt 2016-04-30 14:43:23.172567805 -0700
@@ -99,7 +99,7 @@
 ENDIF()
 IF(GDAL_FOUND)
     ADD_SUBDIRECTORY(gdal)
-    ADD_SUBDIRECTORY(ogr)
+    #ADD_SUBDIRECTORY(ogr)
 ENDIF()
 IF(GTA_FOUND AND OSG_CPP_EXCEPTIONS_AVAILABLE)
     ADD_SUBDIRECTORY(gta)
