diff -crN OpenSceneGraph-3.0.1_org/src/CMakeLists.txt OpenSceneGraph-3.0.1/src/CMakeLists.txt
*** OpenSceneGraph-3.0.1_org/src/CMakeLists.txt	2013-07-12 12:02:46.000000000 -0700
--- OpenSceneGraph-3.0.1/src/CMakeLists.txt	2013-07-12 15:24:02.000000000 -0700
***************
*** 15,33 ****
          osgGA 
          osgText 
          osgViewer
-         osgAnimation 
          osgFX
          osgManipulator
-         osgParticle
-         osgPresentation
-         osgShadow
-         osgSim
          osgTerrain
          osgWidget
-         osgVolume
-         osgWrappers/serializers
-         osgWrappers/deprecated-dotosg
-         osgPlugins
      )
  
      ADD_SUBDIRECTORY(${mylibfolder})
--- 15,24 ----
