diff -crN OpenSceneGraph-3.0.1_org/src/OpenThreads/common/Atomic.cpp OpenSceneGraph-3.0.1/src/OpenThreads/common/Atomic.cpp
*** OpenSceneGraph-3.0.1_org/src/OpenThreads/common/Atomic.cpp	2013-07-12 12:02:46.000000000 -0700
--- OpenSceneGraph-3.0.1/src/OpenThreads/common/Atomic.cpp	2013-07-12 16:14:24.000000000 -0700
***************
*** 143,149 ****
  AtomicPtr::assign(void* ptrNew, const void* const ptrOld)
  {
  #if defined(_OPENTHREADS_ATOMIC_USE_GCC_BUILTINS)
!     return __sync_bool_compare_and_swap(&_ptr, ptrOld, ptrNew);
  #elif defined(_OPENTHREADS_ATOMIC_USE_WIN32_INTERLOCKED)
      return ptrOld == InterlockedCompareExchangePointer((PVOID volatile*)&_ptr, (PVOID)ptrNew, (PVOID)ptrOld);
  #elif defined(_OPENTHREADS_ATOMIC_USE_BSD_ATOMIC)
--- 143,149 ----
  AtomicPtr::assign(void* ptrNew, const void* const ptrOld)
  {
  #if defined(_OPENTHREADS_ATOMIC_USE_GCC_BUILTINS)
!     return __sync_bool_compare_and_swap(&_ptr, (void*)ptrOld, ptrNew);
  #elif defined(_OPENTHREADS_ATOMIC_USE_WIN32_INTERLOCKED)
      return ptrOld == InterlockedCompareExchangePointer((PVOID volatile*)&_ptr, (PVOID)ptrNew, (PVOID)ptrOld);
  #elif defined(_OPENTHREADS_ATOMIC_USE_BSD_ATOMIC)
