--- SuiteSparse/SuiteSparse_config/SuiteSparse_config.mk	2018-10-12 10:27:01.655305384 -0700
+++ SuiteSparse/SuiteSparse_config/SuiteSparse_config.mk	2018-10-12 10:24:44.159632354 -0700
@@ -119,7 +119,7 @@
 # naming the BLAS and LAPACK library (*.a or *.so) files.
 
 # This is probably slow ... it might connect to the Standard Reference BLAS:
-# BLAS = -lblas -lgfortran
+  BLAS = -lblas -lgfortran
   LAPACK = -llapack
 
 # MKL 
@@ -131,7 +131,7 @@
 # LAPACK =
 
 # OpenBLAS
-  BLAS = -lopenblas
+#  BLAS = -lopenblas
 # LAPACK = 
 
 # NOTE: this next option for the "Goto BLAS" has nothing to do with a "goto"
