diff -uN TheiaSfM-0.5/applications/CMakeLists.txt TheiaSfM-0.5/applications_new/CMakeLists.txt
--- TheiaSfM-0.5/applications/CMakeLists.txt	2015-11-25 16:53:06.000000000 -0800
+++ TheiaSfM-0.5/applications_new/CMakeLists.txt	2015-11-25 16:56:41.000000000 -0800
@@ -90,2 +90,7 @@
  add_executable(create_reconstruction_from_strecha_dataset create_reconstruction_from_strecha_dataset.cc)
  target_link_libraries(create_reconstruction_from_strecha_dataset theia ${GFLAGS_LIBRARIES} ${GLOG_LIBRARIES})
+
+ADD_EXECUTABLE(export_reconstruction_to_vw export_reconstruction_to_vw.cc)
+TARGET_LINK_LIBRARIES(export_reconstruction_to_vw theia ${GFLAGS_LIBRARIES} ${GLOG_LIBRARIES})
+
+install (TARGETS build_reconstruction export_reconstruction_to_vw view_reconstruction DESTINATION bin)
diff -uN TheiaSfM-0.5/applications/export_reconstruction_to_vw.cc TheiaSfM-0.5/applications_new/export_reconstruction_to_vw.cc
--- TheiaSfM-0.5/applications/export_reconstruction_to_vw.cc	1969-12-31 16:00:00.000000000 -0800
+++ TheiaSfM-0.5/applications_new/export_reconstruction_to_vw.cc	2015-11-25 15:30:38.000000000 -0800
@@ -0,0 +1,189 @@
+// Copyright (C) 2015 The Regents of the University of California (Regents).
+// All rights reserved.
+//
+// Redistribution and use in source and binary forms, with or without
+// modification, are permitted provided that the following conditions are
+// met:
+//
+//     * Redistributions of source code must retain the above copyright
+//       notice, this list of conditions and the following disclaimer.
+//
+//     * Redistributions in binary form must reproduce the above
+//       copyright notice, this list of conditions and the following
+//       disclaimer in the documentation and/or other materials provided
+//       with the distribution.
+//
+//     * Neither the name of The Regents or University of California nor the
+//       names of its contributors may be used to endorse or promote products
+//       derived from this software without specific prior written permission.
+//
+// THIS SOFTWARE IS PROVIDED BY THE COPYRIGHT HOLDERS AND CONTRIBUTORS "AS IS"
+// AND ANY EXPRESS OR IMPLIED WARRANTIES, INCLUDING, BUT NOT LIMITED TO, THE
+// IMPLIED WARRANTIES OF MERCHANTABILITY AND FITNESS FOR A PARTICULAR PURPOSE
+// ARE DISCLAIMED. IN NO EVENT SHALL THE COPYRIGHT HOLDERS OR CONTRIBUTORS BE
+// LIABLE FOR ANY DIRECT, INDIRECT, INCIDENTAL, SPECIAL, EXEMPLARY, OR
+// CONSEQUENTIAL DAMAGES (INCLUDING, BUT NOT LIMITED TO, PROCUREMENT OF
+// SUBSTITUTE GOODS OR SERVICES; LOSS OF USE, DATA, OR PROFITS; OR BUSINESS
+// INTERRUPTION) HOWEVER CAUSED AND ON ANY THEORY OF LIABILITY, WHETHER IN
+// CONTRACT, STRICT LIABILITY, OR TORT (INCLUDING NEGLIGENCE OR OTHERWISE)
+// ARISING IN ANY WAY OUT OF THE USE OF THIS SOFTWARE, EVEN IF ADVISED OF THE
+// POSSIBILITY OF SUCH DAMAGE.
+//
+// Please contact the author of this library if you have any questions.
+// Author: Chris Sweeney (cmsweeney@cs.ucsb.edu)
+
+#include <glog/logging.h>
+#include <gflags/gflags.h>
+#include <theia/theia.h>
+
+
+#include <stlplus3/file_system.hpp>
+
+#include <fstream>  // NOLINT
+
+DEFINE_string(reconstruction, "", "Theia Reconstruction file.");
+DEFINE_string(images, "",
+              "A filepath wildcard specifying all images that were used in the "
+              "reconstruction.");
+DEFINE_string(output_directory, "",
+              "A directory to store the necessary pmvs files.");
+DEFINE_int32(num_threads, 1, "Number of threads used in PMVS.");
+
+
+
+/*
+bool GetFilepathsFromWildcardDEBUG(
+    const std::string& filepath_with_wildcard,
+    std::vector<std::string>* filepaths) {
+  CHECK_NOTNULL(filepaths)->clear();
+
+  const std::string folder = stlplus::folder_part(filepath_with_wildcard);
+  if (!stlplus::folder_exists(folder)) {
+    VLOG(2) << "Input folder does not exist:" << folder;
+    return false;
+  }
+
+  const std::string filename_part =
+      stlplus::filename_part(filepath_with_wildcard);
+
+  std::cout << "folder        = " << folder << std::endl;
+  std::cout << "filename_part = " << filename_part << std::endl;
+
+  const bool kReturnSubfolders = false;
+  const bool kReturnFiles = true;
+  std::vector<std::string> image_filenames = stlplus::folder_wildcard(
+      folder, filename_part, kReturnSubfolders, kReturnFiles);
+
+  filepaths->resize(image_filenames.size());
+  for (int i = 0; i < filepaths->size(); i++) {
+    std::cout << "found file: " << image_filenames[i] << std::endl;
+    filepaths->at(i) = stlplus::create_filespec(folder, image_filenames[i]);
+  }
+
+  if (filepaths->size() == 0) {
+    VLOG(2) << "No files matched the input.";
+  }
+
+  return true;
+}
+*/
+
+void CreateDirectoryIfDoesNotExist(const std::string& directory) {
+  if (!theia::DirectoryExists(directory)) {
+    CHECK(theia::CreateNewDirectory(directory))
+        << "Could not create the directory: " << directory;
+  }
+}
+
+int WriteCamerasForVW(const theia::Reconstruction& reconstruction) {
+
+  std::vector<std::string> image_files;
+  CHECK(theia::GetFilepathsFromWildcard(FLAGS_images, &image_files))
+      << "Could not find images that matched the filepath: " << FLAGS_images
+      << ". NOTE that the ~ filepath is not supported.";
+  CHECK_GT(image_files.size(), 0) << "No images found in: " << FLAGS_images;
+
+  // Format for printing eigen matrices.
+  const Eigen::IOFormat unaligned(Eigen::StreamPrecision, Eigen::DontAlignCols, " ", " ");
+
+  int current_image_index = 0;
+  for (int i = 0; i < image_files.size(); i++) {
+    // Fetch the view ID for this image
+    std::string image_name;
+    CHECK(theia::GetFilenameFromFilepath(image_files[i], true, &image_name));
+    const theia::ViewId view_id = reconstruction.ViewIdFromName(image_name);
+    if (view_id == theia::kInvalidViewId) {
+      continue;
+    }
+
+    LOG(INFO) << "Exporting parameters for image: " << image_name;
+
+    // Generate file name for this image
+
+    // Write the camera projection matrix.
+    const std::string text_path = stlplus::create_filespec(FLAGS_output_directory, 
+                                                           image_name + ".tsai");
+    LOG(INFO) << "Writing file: " << text_path;
+    //const std::string txt_file = theia::StringPrintf(
+    //    "%s/%08d.txt", FLAGS_output_directory.c_str(), current_image_index);
+    
+    // Get camera information
+    const theia::Camera camera = reconstruction.View(view_id)->Camera();
+    
+    // Write the pinhole camera model to an ASP compatible text file
+    std::ofstream ofs(text_path);
+    
+    // Focal length x then y
+    ofs << "fu = " << camera.FocalLength() << std::endl;
+    ofs << "fv = " <<camera.FocalLength() << std::endl;
+    
+    // Center point x then y
+    ofs << "cu = " << camera.PrincipalPointX() << std::endl;
+    ofs << "cv = " << camera.PrincipalPointY() << std::endl;
+    
+    // Some pixel direction vectors that VW expects
+    ofs << "u_direction = 1  0  0" << std::endl;
+    ofs << "v_direction = 0  1  0" << std::endl;
+    ofs << "w_direction = 0  0  1" << std::endl;
+    
+    // Location C
+    ofs << "C = " << camera.GetPosition().format(unaligned) << std::endl;
+    
+    // Rotation matrix R
+    ofs << "R = " << camera.GetOrientationAsRotationMatrix().transpose().format(unaligned) << std::endl;
+    
+    // Distortion parameters
+    ofs << "k1 = " << camera.RadialDistortion1() << std::endl;
+    ofs << "k2 = " << camera.RadialDistortion2() << std::endl;
+    ofs << "p1 = 0" << std::endl;
+    ofs << "p2 = 0" << std::endl;
+    
+    ofs.close();
+
+    ++current_image_index;
+  } // End loop through images
+
+  return current_image_index;
+}
+
+int main(int argc, char* argv[]) {
+  google::InitGoogleLogging(argv[0]);
+  THEIA_GFLAGS_NAMESPACE::ParseCommandLineFlags(&argc, &argv, true);
+
+  // Set up output directories.
+  CreateDirectoryIfDoesNotExist(FLAGS_output_directory);
+
+  theia::Reconstruction reconstruction;
+  CHECK(theia::ReadReconstruction(FLAGS_reconstruction, &reconstruction))
+      << "Could not read Reconstruction files.";
+
+  /*const int num_cameras = */WriteCamerasForVW(reconstruction);
+  //WritePMVSOptions(FLAGS_output_directory, num_cameras);
+
+  //const std::string lists_file = FLAGS_output_directory + "/list.txt";
+  //const std::string bundle_file =
+  //    FLAGS_output_directory + "/bundle.rd.out";
+  //CHECK(theia::WriteBundlerFiles(reconstruction, lists_file, bundle_file));
+
+  return 0;
+}
