--- boost_1_49_0/Jamroot.orig	2012-02-24 20:38:03.995207243 -0500
+++ boost_1_49_0/Jamroot	2012-02-24 20:38:33.542227242 -0500
@@ -377,7 +377,7 @@
         if $(layout) = versioned
         {
             result = [ common.format-name
-                <base> <toolset> <threading> <runtime> -$(BOOST_VERSION_TAG)
+                <base> <threading> <runtime> -$(BOOST_VERSION_TAG)
                 -$(BUILD_ID)
                 : $(name) : $(type) : $(property-set) ] ;
         }        
