diff -urp boost_1_44_0/tools/build/v2/tools/builtin.jam b/tools/build/v2/tools/builtin.jam
--- boost_1_44_0/tools/build/v2/tools/builtin.jam	2010-04-20 08:05:14.000000000 -0400
+++ b/tools/build/v2/tools/builtin.jam	2012-02-24 18:06:02.585302042 -0500
@@ -107,7 +107,7 @@ feature.feature stdlib             : nat
 feature.feature link               : shared static   : propagated ;
 feature.feature runtime-link       : shared static   : propagated ;
 feature.feature runtime-debugging  : on off          : propagated ;
-feature.feature optimization       : off speed space : propagated ;
+feature.feature optimization       : off speed space none : propagated ;
 feature.feature profiling          : off on          : propagated ;
 feature.feature inlining           : off on full     : propagated ;
 feature.feature threading          : single multi    : propagated ;
@@ -120,7 +120,7 @@ feature.feature asynch-exceptions  : off
 # Whether all extern "C" functions are considered nothrow by default.
 feature.feature extern-c-nothrow   : off on          : propagated ;
 
-feature.feature debug-symbols      : on off          : propagated ;
+feature.feature debug-symbols      : on off none     : propagated ;
 # Controls whether the binary should be stripped -- that is have
 # everything not necessary to running removed. This option should
 # not be very often needed. Also, this feature will show up in
Only in b/tools/build/v2/tools: builtin.jam~
