--- boost-1.39.0/tools/wave/cpp.cpp	2009-05-07 19:26:09.033022587 -0500
+++ boost-1.39.0/tools/wave/cpp.cpp	2009-05-07 19:26:43.659680872 -0500
@@ -799,7 +799,7 @@
 
     // add include directories to the system include search paths
         if (vm.count("sysinclude")) {
-        vector<std::string> syspaths = vm["sysinclude"].as<vector<string> >();
+        vector<std::string> syspaths = vm["sysinclude"].as<vector<std::string> >();
 
             vector<std::string>::const_iterator end = syspaths.end();
             for (vector<std::string>::const_iterator cit = syspaths.begin(); 
@@ -836,7 +836,7 @@
     
     // add additional defined macros 
         if (vm.count("define")) {
-            vector<std::string> const &macros = vm["define"].as<vector<string> >();
+            vector<std::string> const &macros = vm["define"].as<vector<std::string> >();
             vector<std::string>::const_iterator end = macros.end();
             for (vector<std::string>::const_iterator cit = macros.begin(); 
                  cit != end; ++cit)
