--- old/boost/function/function_template.hpp	2008-11-17 22:59:03.000000000 -0600
+++ new/boost/function/function_template.hpp	2008-11-17 21:53:00.000000000 -0600
@@ -950,10 +950,10 @@
           f.vtable->manager(f.functor, this->functor,
                             boost::detail::function::move_functor_tag);
 		  f.vtable = 0;
-#if !defined(BOOST_NO_EXCEPTIONS)      
         } else {
           clear();
         }
+#if !defined(BOOST_NO_EXCEPTIONS)
       } catch (...) {
         vtable = 0;
         throw;
