From 929c49c005d7d7ea238f454bfa772d45f64da581 Mon Sep 17 00:00:00 2001
From: Mike Lundy <mike@fluffypenguin.org>
Date: Sat, 17 Oct 2009 18:11:00 -0700
Subject: [PATCH] override dlopen search methods for osg

---
 src/osgDB/FileUtils.cpp |   27 ++++++++++++++++++++++++++-
 1 files changed, 26 insertions(+), 1 deletions(-)

diff --git a/src/osgDB/FileUtils.cpp b/src/osgDB/FileUtils.cpp
index a0c0315..857562b 100644
--- a/src/osgDB/FileUtils.cpp
+++ b/src/osgDB/FileUtils.cpp
@@ -34,6 +34,7 @@
 #else // unix
 
 #if defined( __APPLE__ )
+    #include <mach-o/dyld.h>
     // I'm not sure how we would handle this in raw Darwin
     // without the AvailablilityMacros.
     #include <AvailabilityMacros.h>
@@ -56,6 +57,8 @@
     #include <unistd.h>
     #include <sys/types.h>
     #include <sys/stat.h>
+    #include <libgen.h>
+    #include <limits.h>
 #endif
 
     // set up _S_ISDIR()
@@ -502,11 +505,31 @@ std::string osgDB::findFileInDirectory(const std::string& fileName,const std::st
 
 static void appendInstallationLibraryFilePaths(osgDB::FilePathList& filepath)
 {
-#ifdef OSG_DEFAULT_LIBRARY_PATH
+#if 0
 
     // Append the install prefix path to the library search path if configured
     filepath.push_back(ADDQUOTES(OSG_DEFAULT_LIBRARY_PATH));
 #endif
+#ifdef __APPLE__
+    const char* filename2 = _dyld_get_image_name(0);
+    char *filename = new char[PATH_MAX];
+    strcpy(filename, filename2);
+
+#else
+    char *filename = new char[PATH_MAX];
+    readlink("/proc/self/exe", filename, PATH_MAX);
+#endif
+
+    if (filename && filename[0] != '\0') {
+        char *copy = new char[PATH_MAX];
+        realpath(filename, copy);
+        std::string top = std::string(dirname(dirname(copy))) + "/lib";
+        delete [] copy;
+        filepath.push_back(top);
+    }
+    else
+        osg::notify(osg::WARN) << "Could not get path to binary!" << std::endl;
+    delete [] filename;
 }
 
 #if defined(WIN32) && !defined(__CYGWIN__)
@@ -921,6 +944,8 @@ static void appendInstallationLibraryFilePaths(osgDB::FilePathList& filepath)
 
         appendInstallationLibraryFilePaths(filepath);
 
+        return;
+
         const std::string OSG_PLUGIN_PATH("/OpenSceneGraph/PlugIns");
         CFURLRef  url;
         CFBundleRef myBundle;
-- 
1.6.4.4

