diff -ruN OpenSceneGraph-2.8.5/src/osgDB/Registry.cpp OpenSceneGraph-2.8.5_mod/src/osgDB/Registry.cpp
--- OpenSceneGraph-2.8.5/src/osgDB/Registry.cpp	2012-08-29 13:53:01.000000000 -0700
+++ OpenSceneGraph-2.8.5_mod/src/osgDB/Registry.cpp	2012-08-29 13:57:42.000000000 -0700
@@ -426,8 +426,12 @@
         setLibraryFilePathList(ptr);
     }
     
-    appendPlatformSpecificLibraryFilePaths(_libraryFilePath);
+    if ( !ptr || strlen(ptr) == 0 ) {
+      notify(FATAL) << "OSG_LIBRARY_PATH was not set! Terminating." << std::endl;
+      std::abort();
+    }
 
+    //appendPlatformSpecificLibraryFilePaths(_libraryFilePath);
 }
 
 
