--- libpointmatcher/CMakeLists.txt	2013-11-19 15:51:21.000000000 -0800
+++ libpointmatcher/CMakeLists.txt	2013-11-19 16:00:47.000000000 -0800
@@ -20,8 +20,12 @@
   set (COMPILER_IS_CLANG TRUE)
 endif (CMAKE_CXX_COMPILER_ID MATCHES "Clang")
 
-# Enable OpenMP
-add_definitions(-fopenmp)
+IF(NOT ${CMAKE_SYSTEM_NAME} MATCHES "Darwin")
+    # Enable OpenMP
+    add_definitions(-fopenmp)
+    # Mac OS X specific code
+    #SET(OperatingSystem "Mac OS X")
+ENDIF(NOT ${CMAKE_SYSTEM_NAME} MATCHES "Darwin")
 
 #if(CMAKE_COMPILER_IS_GNUCXX OR COMPILER_IS_CLANG)
 #	# gcc checks
@@ -151,7 +151,12 @@
 if (SHARED_LIBS)
 	add_library(pointmatcher SHARED ${POINTMATCHER_SRC})
 	install(TARGETS pointmatcher LIBRARY DESTINATION lib)
-        target_link_libraries(pointmatcher ${yaml-cpp_LIBRARIES} ${NABO_LIBRARY} ${Boost_LIBRARIES} "-fopenmp")
+        IF(${CMAKE_SYSTEM_NAME} MATCHES "Darwin")
+            target_link_libraries(pointmatcher ${yaml-cpp_LIBRARIES} ${NABO_LIBRARY} ${Boost_LIBRARY_DIRS} ${Boost_LIBRARY_DIR})
+        ELSE(${CMAKE_SYSTEM_NAME} MATCHES "Darwin")
+            #Link against openmp
+            target_link_libraries(pointmatcher ${yaml-cpp_LIBRARIES} ${NABO_LIBRARY} ${Boost_LIBRARY_DIRS} ${Boost_LIBRARY_DIR} "-fopenmp")
+        ENDIF(${CMAKE_SYSTEM_NAME} MATCHES "Darwin")
 else(SHARED_LIBS)
 	add_library(pointmatcher ${POINTMATCHER_SRC})
 	install(TARGETS pointmatcher ARCHIVE DESTINATION lib)
