diff -ur isis/apps/hideal2pds.dir/hideal2pds.cpp isis_new/apps/hideal2pds.dir/hideal2pds.cpp
--- isis/apps/hideal2pds.dir/hideal2pds.cpp	2013-03-15 11:35:41.000000000 -0700
+++ isis_new/apps/hideal2pds.dir/hideal2pds.cpp	2013-08-08 13:33:02.481595593 -0700
@@ -443,7 +443,7 @@
   // mosaic input cube.
   
   // Add NaifKeywords Object values to the ROOT object
-  QString radiiName = "BODY" + QString(cam->naifBodyCode()) + "_RADII";
+  QString radiiName = "BODY" + QString((int)cam->naifBodyCode()) + "_RADII";
   PvlObject naifKeywordGroup = cam->getStoredNaifKeywords();
 
   if (naifKeywordGroup.hasKeyword(radiiName)) {
diff -ur isis/include/IString.h isis_new/include/IString.h
--- isis/include/IString.h	2013-08-08 13:40:30.587920863 -0700
+++ isis_new/include/IString.h	2013-08-08 13:33:05.765830252 -0700
@@ -42,6 +42,10 @@
   QString toString(char);
   QString toString(const int &);
   QString toString(const unsigned int &);
+#if __x86_64__ || __ppc64__
+#else
+  QString toString(const long &);
+#endif  
   QString toString(const BigInt &);
   QString toString(double, int precision = 14);
 
diff -ur isis/src/Core/IString/IString.cpp isis_new/src/Core/IString/IString.cpp
--- isis/src/Core/IString/IString.cpp	2013-01-04 10:02:32.000000000 -0800
+++ isis_new/src/Core/IString/IString.cpp	2013-08-08 13:32:43.956162460 -0700
@@ -248,6 +248,12 @@
    * @param intToConvert Integer value to be converted to a QString.
    * @return string Converted QString. 
    */
+#if __x86_64__ || __ppc64__
+#else
+ QString toString(const long &intToConvert) {
+    return QString::number(intToConvert);
+ }
+#endif
   QString toString(const int &intToConvert) {
     return QString::number(intToConvert);
   }
