diff -ur isis/apps/hifringe.dir/hifringe.cpp isis_mod/apps/hifringe.dir/hifringe.cpp
--- isis/apps/hifringe.dir/hifringe.cpp	2013-03-15 11:35:41.000000000 -0700
+++ isis_mod/apps/hifringe.dir/hifringe.cpp	2013-12-07 10:39:24.000000000 -0800
@@ -90,9 +90,10 @@
     }
   }
 
-  Statistics sections[numSections][2];
+  std::vector<std::vector<Statistics> >
+    sections(numSections, std::vector<Statistics>(2) );
   Statistics leftTotal, rightTotal;
-  int sectionStarts[numSections];
+  std::vector<int> sectionStarts(numSections);
   sectionStarts[0] = 0;
   for(int i = 1 ; i < numSections - 1 ; i ++) {
     sectionStarts[i] = (totalLines / numSections) * i;
